### PR TITLE
Add code for reading a pclean/model5 schema

### PR DESCRIPTION
**Pull request by ThomasColthurst**
Includes the schema definition and a general purpose tokenizer.

### CODE REVIEW

File: cxx/pclean/io.cc [hunk 1]
```diff
@@ -0,0 +1,284 @@
+
```
Comment (emilyfertig) [inline cxx/pclean/io.cc:1]: wdyt of using absl::Status here? https://abseil.io/docs/cpp/guides/status

**Review by emilyfertig — APPROVED** [review]
LG -- is this mostly ported from Julia PClean?

**Comment by ThomasColthurst** [comment]
> LG -- is this mostly ported from Julia PClean?

Nope; Julia PClean defines the classes and query in Julia code.

File: cxx/pclean/io.cc (hunk 1 above)
Comment (ThomasColthurst) [reply cxx/pclean/io.cc:1]: Mildly against.  The absl::Status codes are all built around RPC errors (https://abseil.io/docs/cpp/guides/status-codes), so they wouldn't really help us distinguish between different parsing errors.  Nor would our expected use cases need to so distinguish.  Plus using the abseil libraries would be another piece of overhead to maintain.

**Comment by emilyfertig** [comment]
I mean, is it mostly rewritten directly from the Julia code?

**Comment by ThomasColthurst** [comment]
It is not.  There is no parsing code in the Julia repo, so rewriting wasn't an option.

**Pull request merged**